samples/client/petstore/scala-sttp/src/main/scala/org/openapitools/client/core/ApiInvoker.scala

```diff
@@ -1,60 +0,0 @@
-/**
- * OpenAPI Petstore
- * This is a sample server Petstore server. For this sample, you can use the api key `special-key` to test the authorization filters.
- *
- * The version of the OpenAPI document: 1.0.0
- * 
- *
- * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
- * https://openapi-generator.tech
- * Do not edit the class manually.
- */
-package org.openapitools.client.core
-
-import org.json4s._
-import sttp.client._
-import sttp.model.StatusCode
-import org.openapitools.client.api.EnumsSerializers
-import sttp.client.json4s.SttpJson4sApi
-import sttp.client.monad.MonadError
-
-class SttpSerializer(implicit val format: Formats = DefaultFormats ++ EnumsSerializers.all ++ Serializers.all,
-                     implicit val serialization: org.json4s.Serialization = org.json4s.jackson.Serialization) extends SttpJson4sApi
-
-class HttpException(val statusCode: StatusCode, val statusText: String, val message: String) extends Exception(s"[$statusCode] $statusText: $message")
-
-object Helpers {
-
-  // Helper to handle Optional header parameters
-  implicit class optionalParams(val request: RequestT[Identity, Either[String, String], Nothing]) extends AnyVal {
-    def header( header: String, optValue: Option[Any]): RequestT[Identity, Either[String, String], Nothing] = {
-      optValue.map( value => request.header(header, value.toString)).getOrElse(request)
-    }
-  }
-
-}
-
-object ApiInvoker {
-
-  /**
-    * Allows request execution without calling apiInvoker.execute(request)
-    * request.result can be used to get a monad wrapped content.
-    *
-    * @param request the apiRequest to be executed
-    */
-  implicit class ApiRequestImprovements[R[_], T](request: RequestT[Identity, Either[ResponseError[Exception], T], Nothing]) {
-
-    def result(implicit backend: SttpBackend[R, Nothing, Nothing]): R[T] = {
```

Comment:
The way sttp-client is normally used (at least, what we've seen on gitter, issue reports etc.) is that people simply call `.send()` with an implicit backend in scope. What kind of functionality would the helper add?

Reply:
I think that helper can be on the app side as error transformations can be different from app to app.
The main idea of that helper is to get successful result directly. For example for `Try` framework it could be something like 
```
val pet: Pet = new PetApi("http://swagger").getPetById(1).result()
```

instead of code like next for every method
```
 new PetApi("http://swagger").getPetById(1).send()
   .flatMap { 
      case Right(pet) => pet 
            doSomething(pet)
      case ...
   }
```

Reply:
For that purpose we've introduced the `separateErrorChannel` option: using sttp's mechanism (response specifications), requests can be modeled to return full error information (so the response will be roughly an `Either[Error, Success]`), or the errors can be represented as failed effects (as the errors are `Throwables`).

Of course people might want to generate code using full error information and then flatten the result anyway. Though here, as you write, it's quite an application-specific thing. I think if I wanted to flatten the results, I'd write an extension method, which would convert `F[Either[Error, Success]]` into a `F[Success]`, e.g.:

```
new PetApi("http://swagger").getPetById(1).send().flattenErrors
```

where

```
implicit class TryOps[E <: Throwable, S](r: Try[Either[E, S]]) {
  def flattenErrors: Try[S] = r.flatMap {
    case Left(e) => Failed(e)
    case Right(s) => Success(s)
  }
}
```

but I think this would belong to the specific app - not in the generated code. Otherwise we should have this in sttp directly :)